report-job-status/action.yml

```diff
@@ -74,18 +99,37 @@ runs:
                 JSON.stringify(steps[key])
             }
           })
+          
+          // Try to find the job ID by its name
+          // In matrix workflows it will work only if exact name was provided
+          // in `job-name` input.
+          for (job of jobContext.jobs) {
+            if (job.name == failedJobName ) {
+              jobId = job.id
```

Comment:
When we get the match, we should use `break` because there is no sense to iterate further.